CoreFoundation/NumberDate.subproj/CFTimeZone.c

```diff
@@ -1149,7 +1347,9 @@ Boolean _CFTimeZoneInit(CFTimeZoneRef timeZone, CFStringRef name, CFDataRef data
         CFStringRef mapping = CFDictionaryGetValue(dict, name);
         if (mapping) {
             name = mapping;
-        } else if (CFStringHasPrefix(name, __tzZoneInfo)) {
+        }
+#if !TARGET_OS_ANDROID
```

Comment:
I think it would be nicer to just fold this into the condition below as `} else if (!TARGET_OS_ANDROID && CFStringHasPrefix(name, __tzZoneInfo))`

Reply:
`__tzZoneInfo` doesn't exist in Android, so even if that will work, the references to the variable will make the compilation fail.

PD: `__tzZoneInfo` used to exist in Android in a previous version of this PR, but it was an empty dummy, which removes a lot of conditional compilation from the rest of the code.

Reply:
Oh, right.  Yeah, that was the desire, to get rid of that path which doesn't really make sense.